src/Mod/Sketcher/Gui/EditModeInformationOverlayCoinConverter.cpp

```diff
@@ -82,167 +86,188 @@ void EditModeInformationOverlayCoinConverter::addToInfoGroup(SoSwitch * sw) {
 
 template < EditModeInformationOverlayCoinConverter::CalculationType calculation >
 void EditModeInformationOverlayCoinConverter::calculate(const Part::Geometry * geometry, [[maybe_unused]] int geoid) {
-    const Part::GeomBSplineCurve *spline = static_cast<const Part::GeomBSplineCurve *>(geometry);
-
-    if constexpr (calculation == CalculationType::BSplineDegree ) {
-        clearCalculation(degree);
-
-        std::vector<Base::Vector3d> poles = spline->getPoles();
 
-        degree.strings.clear();
-        degree.positions.clear();
+    if constexpr (calculation == CalculationType::ArcCircleHelper ) {
+    	const Part::GeomArcOfCircle *arc = static_cast<const Part::GeomArcOfCircle *>(geometry);
+        clearCalculation(circleHelper);
 
-        Base::Vector3d midp = Base::Vector3d(0,0,0);
-
-        for (auto val : poles)
-            midp += val;
-
-        midp /= poles.size();
-
-        degree.strings.emplace_back(std::to_string(spline->getDegree()));
-        degree.positions.emplace_back(midp);
-    }
-    else if constexpr (calculation == CalculationType::BSplineControlPolygon ) {
+        Base::Vector3d center = arc->getCenter();
+        double radius = arc->getRadius();
+        Part::GeomCircle circle;
+        circle.setRadius(radius);
+        circle.setCenter(center);
 
-        clearCalculation(controlPolygon);
+        const int ndiv = drawingParameters.curvedEdgeCountSegments;
 
-        std::vector<Base::Vector3d> poles = spline->getPoles();
-
-        controlPolygon.coordinates.clear();
-        controlPolygon.indices.clear();
-
-        size_t nvertices;
+        circleHelper.coordinates.reserve(ndiv);
+        for (int i = 0; i < ndiv; i++) {
+        	double param = i * std::atan(1) * 8 / ndiv;
+        	circleHelper.coordinates.emplace_back(circle.value(param));
+        }

```

Comment:
Do we really need to use something tessellated ? Isn't Coin able to draw arc of circles directly ?

Reply:
I tried this for B-splines once. Didn't have much success. I did see that being done in python and can dig out the forum. If we change this we should consider changing everything else as well in a separate PR.

Reply:
This one is "only" for information layer, it is not the regular one for geometry. 

Anyway, it is not an straight-forward change. For sure in a separate PR.